diff --git a/examples/rps/rsh/rps.rsh b/examples/rps/rsh/rps.rsh
index 56f3167..c0d3d8c 100644
--- a/examples/rps/rsh/rps.rsh
+++ b/examples/rps/rsh/rps.rsh
@@ -79,7 +79,7 @@ function main() {
       return [0, 2 * wagerAmount]; }
     else {
       return [wagerAmount, wagerAmount]; } })();
-  transfer(escrowAmount + getsA).to(A);
+  transfer(/*escrowAmount +*/ getsA).to(A);
   transfer(getsB).to(B);
   commit();
 
