diff --git a/examples/rps/rsh/rps.rsh b/examples/rps/rsh/rps.rsh
index 56f3167..d44620f 100644
--- a/examples/rps/rsh/rps.rsh
+++ b/examples/rps/rsh/rps.rsh
@@ -58,7 +58,7 @@ function main() {
 
   A.only(() => {
     const saltA = declassify(_saltA);
-    const handA = declassify(_handA);
+    const handA = declassify(/*_handA*/getHand());
     interact.reveals(showHand(handB)); });
   A.publish(saltA, handA)
