diff --git a/idb/postgres/internal/encoding/encode.go b/idb/postgres/internal/encoding/encode.go
index f1fc041..5df7774 100644
--- a/idb/postgres/internal/encoding/encode.go
+++ b/idb/postgres/internal/encoding/encode.go
@@ -112,9 +112,35 @@ func convertEvalDelta(evalDelta types.EvalDelta) types.EvalDelta {
 	evalDelta.Logs = removeNonUTF8Chars(evalDelta.Logs)
 	evalDelta.GlobalDelta = convertStateDelta(evalDelta.GlobalDelta)
 	evalDelta.LocalDeltas = convertLocalDeltas(evalDelta.LocalDeltas)
+	evalDelta.InnerTxns = convertInnerTxns(evalDelta.InnerTxns)
 	return evalDelta
 }
 
+func convertInnerTxns(stxns []types.SignedTxnWithAD) []types.SignedTxnWithAD {
+	if stxns == nil {
+		return nil
+	}
+
+	res := make([]types.SignedTxnWithAD, len(stxns))
+	for i, stxn := range stxns {
+		res[i] = convertInnerTxn(stxn)
+	}
+	return res
+}
+
+func removeNonUTF8Chars_AP(params sdk_types.AssetParams) sdk_types.AssetParams {
+  params.AssetName = printableUTF8OrEmpty(params.AssetName)
+  params.UnitName = printableUTF8OrEmpty(params.UnitName)
+  params.URL = printableUTF8OrEmpty(params.URL)
+  return params
+}
+
+func convertInnerTxn(stxn types.SignedTxnWithAD) types.SignedTxnWithAD {
+  stxn = convertSignedTxn(stxn)
+  stxn.Txn.AssetParams = removeNonUTF8Chars_AP(stxn.Txn.AssetParams)
+  return stxn
+}
+
 func convertTransaction(txn types.Transaction) transaction {
 	return transaction{
 		Transaction:         txn,
@@ -122,8 +148,19 @@ func convertTransaction(txn types.Transaction) transaction {
 	}
 }
 
-func convertSignedTxnWithAD(stxn types.SignedTxnWithAD) signedTxnWithAD {
+func convertApplyData(ad types.ApplyData) types.ApplyData {
+	ad.EvalDelta = convertEvalDelta(ad.EvalDelta)
+	return ad
+}
+
+func convertSignedTxn(stxn types.SignedTxnWithAD) types.SignedTxnWithAD {
+	stxn.ApplyData = convertApplyData(stxn.ApplyData)
 	stxn.EvalDelta = convertEvalDelta(stxn.EvalDelta)
+	return stxn
+}
+
+func convertSignedTxnWithAD(stxn types.SignedTxnWithAD) signedTxnWithAD {
+	stxn = convertSignedTxn(stxn)
 	return signedTxnWithAD{
 		SignedTxnWithAD: stxn,
 		TxnOverride:     convertTransaction(stxn.Txn),
