--- /conflux-rust/core/src/spec/genesis.rs 	2021-10-01 08:59:15.000000000 -0300
+++ /patch/genesis-v1.1.5-reach.rs	2021-10-01 08:58:39.000000000 -0300
@@ -85,10 +85,10 @@
 
     let mut accounts: HashMap<Address, U256> = HashMap::new();
     let balance =
-        U256::from_dec_str("10000000000000000000000").map_err(|e| {
+        U256::from_dec_str("100000000000000000000000000").map_err(|e| {
             format!(
                 "failed to parse balance: value = {}, error = {:?}",
-                "10000000000000000000000", e
+                "100000000000000000000000000", e
             )
         })?;
     for line in buffered.lines() {
